packages/core/src/internal/execution/nonce-management/get-nonce-sync-messages.ts

```diff
@@ -279,3 +242,43 @@ function createMapFromSenderToNonceAndTransactions(
 
   return pendingTransactionsPerAccount;
 }
+
+/**
+ * Scan the futures for upcoming account usage, add them to the list,
+ * including the default sender if there are any undefined froms
+ */
+function _resolveFutureSenders(
+  ignitionModule: IgnitionModule<string, string, IgnitionModuleResult<string>>,
+  accounts: string[],
+  defaultSender: string,
+  exStateIds: string[]
+): string[] {
+  const futures = getFuturesFromModule(ignitionModule);
+
+  const senders: string[] = futures
+    .filter((f) => !exStateIds.includes(f.id))
+    .map((f) => _pickFrom(f, accounts, defaultSender))
+    .filter((x): x is string => x !== null);
+
+  return uniq(senders);
+}
+
+function _pickFrom(
+  future: Future,
+  accounts: string[],
+  defaultSender: string
+): string | null {
+  if (isNamedContractAtFuture(future)) {
```

Comment:
I took this pattern as opposed to a sophisticated filter higher up as it gives nicer typing under conditions where we change futures that breaks the `f.from` here.